### PR TITLE
Feat: Update root component props usage

### PR DESCRIPTION
**Pull request by josenriagu**
## Description

- update root component props usage across apps

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: tests/src/data-generator/integrations.ts [hunk 1]
```diff
@@ -34,13 +35,20 @@ const log: any = {
     /*  */
   },
 };
-export const genAppProps = (): RootComponentProps => ({
+export const genAppProps = (): UseRootComponentReturn => ({
```
Comment (SeverS) [inline tests/src/data-generator/integrations.ts]:
```suggestion
export const genAppProps = (): ReturnType<typeof useRootComponentProps> => ({
```

File: typings/src/ui/root-component.ts [hunk 2]
```diff
@@ -32,3 +32,9 @@ export interface RootComponentProps {
 export interface RootExtensionProps extends RootComponentProps {
   extensionData: EventDataTypes;
 }
+
+// @Todo: fix type
+export interface UseRootComponentReturn extends RootComponentProps {
```
Comment (SeverS) [inline typings/src/ui/root-component.ts]: no need for this type!

File: ui/apps/akasha-verse/src/components/App.tsx [hunk 3]
```diff
@@ -1,14 +1,19 @@
 import * as React from 'react';
 import { I18nextProvider } from 'react-i18next';
-import { RootComponentProps } from '@akashaorg/typings/ui';
+
```
Comment (SeverS) [inline ui/apps/akasha-verse/src/components/App.tsx]: remove empty lines between imports please

File: ui/apps/akasha-verse/src/components/pages/explore-page.tsx [hunk 4]
```diff
@@ -8,12 +8,9 @@ import AppList from '@akashaorg/design-system-components/lib/components/AppList'
 import Text from '@akashaorg/design-system-core/lib/components/Text';
 import { useTranslation } from 'react-i18next';
 import { APP_EVENTS } from '@akashaorg/typings/sdk';
-import { RootComponentProps } from '@akashaorg/typings/ui';
 import { GetAppsQuery, GetAppsByIdQuery } from '@akashaorg/typings/sdk/graphql-operation-types-new';
-import { IntegrationReleaseInfo } from '@akashaorg/typings/sdk/graphql-types';
-import { INFO } from '../../routes';
 
-export interface IExplorePage extends RootComponentProps {
+export interface IExplorePage {
```
Comment (SeverS) [inline ui/apps/akasha-verse/src/components/pages/explore-page.tsx]: convert to type

File: tests/src/data-generator/integrations.ts (hunk 1 above)
Comment (josenriagu) [reply tests/src/data-generator/integrations.ts]: we tried this but it caused some cyclic dependency error, because it would need to have hooks package as dependency, whereas hooks package already imports tests package as deps

File: ui/apps/akasha-verse/src/components/pages/apps-page.tsx [hunk 5]
```diff
@@ -4,9 +4,8 @@ import Stack from '@akashaorg/design-system-core/lib/components/Stack';
 import Text from '@akashaorg/design-system-core/lib/components/Text';
 import AppList from '@akashaorg/design-system-components/lib/components/AppList';
 import { useTranslation } from 'react-i18next';
-import { RootComponentProps } from '@akashaorg/typings/ui';
 
-const AppsPage: React.FC<RootComponentProps> = () => {
```
Comment (SeverS) [inline ui/apps/akasha-verse/src/components/pages/apps-page.tsx:9]: you can still keep the typing for React components:
```
AppsPage: React.FC<unknown>
```
or
```
AppsPage: React.FC<never>
```

File: ui/apps/akasha-verse/src/components/pages/explore-page.tsx (hunk 4 above)
Comment (josenriagu) [reply ui/apps/akasha-verse/src/components/pages/explore-page.tsx]: Alright. There are lots of other places where interface was used. Converting all might mean greater number of files being changed. Do I do it on this same PR?

Comment (SeverS) [reply ui/apps/akasha-verse/src/components/pages/explore-page.tsx]: we'll do it in steps, whenever you change or update one, also change it to type

Comment (josenriagu) [reply ui/apps/akasha-verse/src/components/pages/explore-page.tsx]: okay, I will scan through only the changed files and adjust accordingly

File: ui/apps/auth-app/src/components/connect/connect-wallet.tsx [hunk 6]
```diff
@@ -65,6 +65,7 @@ const ConnectWallet: React.FC<IConnectWalletProps> = props => {
     return () => {
       changedNetworkUnsubscribe();
     };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (SeverS) [inline ui/apps/auth-app/src/components/connect/connect-wallet.tsx]: remove this comment please... let's fix hook dependencies as we go through

File: ui/apps/akasha-verse/src/components/pages/info-page.tsx [hunk 7]
```diff
@@ -40,26 +34,26 @@ const InfoPage: React.FC<RootComponentProps> = props => {
...
+  //     window.open(`https://etherscan.io/address/${ethAddress}`, '_blank', 'noreferrer noopener');
+  //   }
+  // };
+
+  // const handleAuthorClick = (author: { pubKey: string }) => {
+  //   navigateTo?.({
+  //     appName: '@akashaorg/app-profile',
+  //     getNavigationUrl: routes => `${routes.rootRoute}/${author.pubKey}`,
+  //   });
+  // };
```
Comment (didd) [inline ui/apps/akasha-verse/src/components/pages/info-page.tsx:56]: Here a todo comment is helpful

File: ui/apps/akasha-verse/src/components/pages/my-widgets-page.tsx [hunk 8]
```diff
@@ -1,51 +1,50 @@
...
+  // const defaultWidgetNamesNormalized = React.useMemo(() => {
+  //   return worldConfig?.defaultWidgets.map(app => {
+  //     if (typeof app === 'string') {
+  //       return {
+  //         name: app,
+  //       };
+  //     }
+  //     return app;
+  //   });
+  // }, [worldConfig.defaultWidgets]);
```
Comment (didd) [inline ui/apps/akasha-verse/src/components/pages/my-widgets-page.tsx:29]: A todo comment is helpful here

File: ui/apps/akasha/src/components/my-feed-page/my-feed-page.tsx [hunk 9]
```diff
@@ -41,7 +44,9 @@ const MyFeedPage: React.FC<MyFeedPageProps & RootComponentProps> = props => {
       },
     },
   );
-  const navigateToModal = React.useRef(props.navigateToModal);
+
+  const _navigateToModal = React.useRef(navigateToModal);
```
Comment (didd) [inline ui/apps/akasha/src/components/my-feed-page/my-feed-page.tsx:48]: It's not part of this PR but I am not sure what the useRef is doing here.

File: ui/apps/articles/src/pages/settings.tsx [hunk 10]
```diff
@@ -52,7 +49,7 @@ const ArticleSettingsPage: React.FC<RootComponentProps> = props => {
       subscribedTopics={selectedTopics}
       uninstallLabel={t('Uninstall')}
       onClickCloseSettings={handleClickCloseSettings}
-      onSearch={handleSearch}
+      onSearch={() => null}
```
Comment (didd) [inline ui/apps/articles/src/pages/settings.tsx:52]: A todo comment is helpful here so that we don't forget about it

**Comment by didd** [comment]
Except for some adjustments the PR looks fine to me

**Review by SeverS — APPROVED** [review]
